### PR TITLE
PPl `flatten` command

### PR DESCRIPTION
**Pull request by lukasz-soszynski-eliatra**
### Description
The `flatten` command is introduced

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/669

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: docs/ppl-lang/ppl-fillnull-command.md
[the remark this answers is not in the document]
Comment (lukasz-soszynski-eliatra) [reply docs/ppl-lang/ppl-fillnull-command.md:1]: Sorry, I forgot to commit the file during work on the `fillnull` command. Should I move the file into a dedicated MR?

[the remark this answers is not in the document]
Comment (lukasz-soszynski-eliatra) [reply docs/ppl-lang/ppl-fillnull-command.md:1]: Here is MR with the `fillnull` command documentation: https://github.com/opensearch-project/opensearch-spark/pull/815

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class PPLLogicalPlanFlattenCommandTranslatorTestSuite
+    extends SparkFunSuite
+    with PlanTest
+    with LogicalPlanTestUtils
+    with Matchers {
+
+  private val planTransformer = new CatalystQueryPlanVisitor()
+  private val pplParser = new PPLSyntaxParser()
+
+  test("test fillnull only field") {
```
Comment (YANG-DB) [inline ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala:29]: @lukasz-soszynski-eliatra plz change the test name to match the ppl query

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala
Comment (YANG-DB) [inline ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala:1]: @lukasz-soszynski-eliatra plz add more tests with different use cases plz

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFlattenITSuite.scala [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    with StreamTest {
+
+  private val testTable = "flint_ppl_test"
+  private val tempFile = Files.createTempFile("jsonTestData", ".json")
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    // Create test table
+    createSimpleNestedJsonContentTable(tempFile, testTable)
```
Comment (YANG-DB) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFlattenITSuite.scala:31]: @lukasz-soszynski-eliatra plz add additional test from other types of tables or use cases:
- [NestesTable](https://github.com/opensearch-project/opensearch-spark/blob/a07f88f86fa384d94e535f99397e8d0d0402bba0/integ-test/src/integration/scala/org/opensearch/flint/spark/FlintSparkSuite.scala#L511)
- [structTable](https://github.com/opensearch-project/opensearch-spark/blob/a07f88f86fa384d94e535f99397e8d0d0402bba0/integ-test/src/integration/scala/org/opensearch/flint/spark/FlintSparkSuite.scala#L487)
- table with multiValue column (Array)

**Review by YANG-DB — CHANGES_REQUESTED** [review]

**Comment by YANG-DB** [comment]
@lukasz-soszynski-eliatra also plz rebase ...

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala
Comment (lukasz-soszynski-eliatra) [reply ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala:1]: done

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFlattenITSuite.scala (hunk 2 above)
Comment (lukasz-soszynski-eliatra) [reply integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLFlattenITSuite.scala]: done

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala (hunk 1 above)
Comment (lukasz-soszynski-eliatra) [reply ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanFlattenCommandTranslatorTestSuite.scala]: done

**Review by YANG-DB — APPROVED** [review]

File: docs/ppl-lang/ppl-flatten-command.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+## PPL `flatten` command
+
+### Description
+Using `flatten` command to flatten a field of type:
+- `struct<?,?>`
+- `array<struct<?,?>>`
```
Comment (LantaoJin) [inline docs/ppl-lang/ppl-flatten-command.md:6]: Are these data_types what we expecting for this command? I thought the expected input is json string. But it's fine for now. We can enhance it later.

File: ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4 [hunk 4]
```diff
@@ -53,6 +53,7 @@ commands
    | renameCommand
    | fillnullCommand
    | fieldsummaryCommand
+   | flattenCommand
    ;
 
 commandName
```
[the remark this answers is not in the document]
Comment (LantaoJin) [reply ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4:59]: LGTM except this^

File: docs/ppl-lang/ppl-flatten-command.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+|-----------|-------------------------------------------------|
+| \_time    | string                                          |
+| bridges   | array\<struct\<length:bigint,name:string\>\>    |
+| city      | string                                          |
+| coor      | struct\<alt:bigint,lat:double,long:double\>     |
+| country   | string                                          |
+#### Data
+| \_time              | bridges                                      | city    | coor                   | country       |
+|---------------------|----------------------------------------------|---------|------------------------|---------------|
+| 2024-09-13T12:00:00 | [{801, Tower Bridge}, {928, London Bridge}]  | London  | {35, 51.5074, -0.1278} | England       |
```
[the remark this answers is not in the document]
Comment (lukasz-soszynski-eliatra) [reply docs/ppl-lang/ppl-flatten-command.md:26]: These data come from the issue https://github.com/opensearch-project/opensearch-spark/issues/669 but were extended.

File: ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4 (hunk 4 above)
[the remark this answers is not in the document]
Comment (lukasz-soszynski-eliatra) [reply ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4:59]: Corrected.

[the remark this answers is not in the document]
Comment (YANG-DB) [reply ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4:59]: > Please add `FLATTEN` keyword in the end of `commandName` section.
[please add here](https://github.com/opensearch-project/opensearch-spark/blob/0a0e2c5d690839f982fa279ac0f19938e551150b/ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4#L83C12-L84C14)

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4 (hunk 4 above)
[the remark this answers is not in the document]
Comment (salyh) [reply ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4:59]: done

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**